### DISCUSSION TITLE
Tab-Nodes-Tree.app is a Tree-Style-Tabs like Chrome Extension

### DISCUSSION THREAD

**Discussion by stefan123t** [piroor/treestyletab#3650] [General]
I just searched again for an incarnation of Tree-Style-Tabs on the Chrome extension web store.
And finally after more than ten years there is [tab-nodes-tree.app](https://www.tab-nodes-tree.app/) which is finally a version which is running in a sidebar and stays open on all the tabs / windows.
Horizontal tabs have become such an anachronism to me, ever since I discovered @piroor's excellent plugin some twenty years ago.
Many thanks for your efforts to provide, support and maintain tree-style-tabs over the past two decades !
Now I finally have an option to use Chrome more frequently too.

The github repo is here: https://github.com/Able-Bear-Software/tab-nodes-tree
And the chrome extension can be downloaded here: https://chromewebstore.google.com/detail/tab-nodes-tree/iglmibjneogoognfbjpmionndflakpeh

Just one bumper, the github repo is only used as a discussion forum, there is no open source code for the extension.
Roy Liao obviously offers a Pro version for ~1$ / month to make a living.

Edit: there is another extension which works very similar to [Tree Style Tabs (Firefox)](https://addons.mozilla.org/en-US/firefox/addon/tree-style-tab/) and [Tab Nodes Tree (Chrome)](https://chromewebstore.google.com/detail/tab-nodes-tree/iglmibjneogoognfbjpmionndflakpeh).
It is called [Tree Tab Sidebar (Chrome)](https://chromewebstore.google.com/detail/tree-tab-sidebar/ifhpbebopidcdemkjmpclgakgihpohie ) by a japanese developer named Kanaru Sato.
I do not know if he has open sourced his Extension and it does not support reordering tabs in the sidebar (yet),

**Comment by SHHSSH** [comment]
> twenty years ago

What the f? I had no idea. This has gotta be an exaggeration?

**Reply by irvinm** [reply]
Well, this repo goes back to late 2010 ... so that is 14 years.  So 20 might be a little bit of an exaggeration. (I don't think it was ever hosted on SourceForge or something like that before Github, but it is possible)

Looks like I entered my first TST bug in 2011. :)

**Reply by piroor** [reply]
The initial version of TST was released at 2007.
https://github.com/piroor/treestyletab/blob/bad6c3c6cec07409605c210502e1d184be3bdb58/history.en.md?plain=1#L3207
And, the initial version with tree like tabs of the Tabbrowser Extensions (the origin of TST) is 2.0.20051116.
https://piro.sakura.ne.jp/xul/tabextensions/history.html.en
Thus my addons to provide tree-like indented tabs is really about 20 years :)

**Reply by stefan123t** [reply]
@piroor oh my bad I thought I used it already since 2004, but 2005-2007 that would be only 17-19 years of TST :smiley_cat:   
Anyway I am deeply indebted to you and your essential plugin, thanks for that !
I recall there was a Mozilla design study and some of the plugins, one called vertical tabs or similar were inspired by that and I had a look at a couple of implementations.
I do not know anymore which one was first, but yours stood out for simplicity and especially the tree structure back then.
The concept of opening tabs in a descendant tab tree immediately caught my attention and I have been sticking with it ever since.

**Comment by stefan123t** [comment]
@piroor I recall that you long harboured the idea of porting it to Chrome, but could not as the browser designs are/were so much different.

I assume a large part of the Tree Style Tabs code to manage the tree structure, etc. to be relevant on both Browsers, 
while there is definitely another big chunk which is only necessary / possible on either of the two Browser architectures. 
But the common code parts should IMHO benefit in clarity of their abstraction from being used by two implementations.

Would you reconsider a port of Tree Style Tabs to Google Chrome now, that we have basic support using sidebar / sidepanel ?

**Reply by piroor** [reply]
There are some blockers:

* TST depends on Firefox specific APIs like https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/sessions/setTabValue to associate tabs and TST-specific state of tabs across browser sessions. TST is designed based on this API, so restructuring everything on cross-browser APIs will take much time.
* I'm still on Firefox and have no plan to move to Chrome. The main motivation of my development is "making my life environment better", so I have very few motivation to develop such a complex addon for a non-daily environment. To be honest this is the largest reason why I don't develop TST for Chrome by myself.

Chrome completely dropped permanent background page support. This kind addon require a permanent process/thread to track tab changes while the sidebar is closed. There are some techniques to simulate such a permanent process with Manifest V3 genuine way, so this point looks not a blocker anymore.
